python/pyarrow/_fs.pyx

```diff
@@ -78,6 +78,12 @@ cdef CFileType _unwrap_file_type(FileType ty) except *:
     assert 0
 
 
+def _file_type_to_string(ty):
+    # Python 3.11 changed str(IntEnum) to return the string representation
+    # of the integer value: https://github.com/python/cpython/issues/94763
+    return f"{ty.__class__.__name__}.{ty._name_}"
```

Comment:
Actually, the change is only for `str(..)`, not for the repr. So then I suppose the above is fine (just checking `FileInfo.type` interactively will still give an informative repr, it's only when printing it that it will now return just an int)

Reply:
I understand that there's no need for a follow up ticket, right?

Reply:
Yes, I think it is fine